libs/solid/quiz/src/lib/components/ranking-question/ranking-question.component.ts

```diff
@@ -44,28 +47,59 @@ export class RankingQuestionComponent implements OnInit, OnChanges {
 
   drop(event: CdkDragDrop<any[]>) {
     moveItemInArray(this.answersList, event.previousIndex, event.currentIndex);
-    console.log(this.answersList);
+    console.log(this.question.answers);
   }
 
   public onShowAnswersClick() {
     this.ShowAnswers = true;
-    let subsequence = false;
+    let hasSubsequence = false;
```

Comment:
Das sieht ziemlich kompliziert aus, aber ich sehe auch ein, dass es die "sichere" Methode ist weil wir uns auf die Reihenfolge de API response nicht unbedingt verlassen können.

Aber Du könntest wahrscheinlich auch mit `Array.forEach` und `Array.filter` arbeiten (ich habe das jetzt nicht ausprobiert). Ich würde vorschlagen, es jetzt so zu lassen, weil der Code so natürlich leicht lesbar ist. 

Vielleicht machen wir dann am Ende noch ein Refactoring und schauen, wo wir die For-Loops noch mit echten JS-Array-Funktionen vereinfachen können.

Ich mache das auch oft, dass ich erst klassische For-Loops aufschreibe, um zu verstehen, was passiert :wink:

Reply:
Wenn alles gut läuft kann ich den Algorithmus später noch mal nachschauen und verfeinern

Reply:
Ja, ich habe ihn noch nicht nachvollzogen, aber wenn er funktioniert muss ich auch nicht wissen, wie ;-)

Reply:
Es wird immer die Positionen von zwei Antworten verglichen, die nebeneinander stehen (weil die kleinste mögliche Subsequence 2 ist). Wenn die eine Subsequence bilden (Position(i) == Position(i+1) - 1) -> werden die beiden Antworten in der `subsequence`-Array gepusht. Sonst wird das nächste Paar verglichen. Es gibt auch ein paar Dinge zu beachten:

1. `subsequence` speichert die am aktuellsten Subsquence. D.h wenn keine neue Antwort gepusht wird, wird `subsequence` in `subsequences` gepusht und auf leer gesetzt.
2. `subsequences` speichert alle gefundene Subsequence, damit wir später identifizieren können, welche am längsten ist.
3. `subsequence` wird nur in `subsequences` gepusht, wenn Elemente darin stehen.
4. Wenn mehrere Paare nebeneinander stehen, die Subsequence bilden, werden nur die neue Antwort in `subsequence` gepusht. 
**Beispiel**: [1,2,3]
1 und 2 ist eine Subsequence -> subsequence=[1,2]
2 und 3 ist eine Subsequence -> subsequence=[1,2,3] statt [1,2,2,3]

Danach suchen wir nach der längsten Subsequence in `subsequences`

P.s: Für später Reference